### PR TITLE
Fixes to the Token Registration dApp

### PR DESCRIPTION
**Pull request by ngotchac**
Small fixes for issues #2195, #2198, #2193, #2197, #2196, #2194 ...

### CODE REVIEW

File: js/src/dapps/tokenreg/Inputs/Text/input-text.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        <Loading size={ 0.3 } />
+      </div>
+    );
+  }
+
+  renderIsValid () {
+    if (this.state.loading || !this.state.valid) return;
+
+    return (
+      <div className={ styles['input-icon'] }>
```
Comment (jacogr) [inline js/src/dapps/tokenreg/Inputs/Text/input-text.js]: I believe styles.inputIcon _may_ do the trick as well.

File: js/src/dapps/tokenreg/Inputs/Text/input-text.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+    this.setValidation(validation);
+  }
+
+  setValidation = (validation) => {
+    const { value } = validation;
+
+    this.setState({ ...validation });
+
+    if (validation.valid) return this.props.onChange(true, value);
```
Comment (jacogr) [inline js/src/dapps/tokenreg/Inputs/Text/input-text.js]: Are you sure the linter doesn't complain here? (Single-line if)

File: js/src/dapps/tokenreg/Tokens/Token/token.js [hunk 3]
```diff
@@ -185,13 +276,14 @@ export default class Token extends Component {
   }
 
   onMetaLookup = () => {
-    let query = this.state.metaQuery;
+    let keyIndex = this.state.metaKeyIndex;
```
Comment (jacogr) [inline js/src/dapps/tokenreg/Tokens/Token/token.js]: let vs const?

File: js/src/dapps/tokenreg/Tokens/actions.js [hunk 4]
```diff
@@ -98,13 +105,33 @@ export const loadToken = (index) => (dispatch, getState) => {
...
+    .then(data => {
+      return getTokenTotalSupply(data.address)
+        .then(totalSupply => {
+          data.totalSupply = totalSupply;
+          return data;
+        });
+    })
+    .then(data => {
+      // If no total supply, must not be a proper token
+      if (data.totalSupply === null) {
```
Comment (jacogr) [inline js/src/dapps/tokenreg/Tokens/actions.js]: does it come back at null or 0? May want to do a eth_getCode on the address first, no code found, not a contract

Comment (ngotchac) [inline js/src/dapps/tokenreg/Tokens/actions.js]: Yep, the `getTokenTotalSupply` method is in fact doing just that, returning `null` if no code, or not a token contract.

File: js/src/dapps/tokenreg/Inputs/Text/input-text.js (hunk 1 above)
Comment (ngotchac) [inline js/src/dapps/tokenreg/Inputs/Text/input-text.js]: Unfortunately, no...

File: js/src/dapps/tokenreg/Inputs/Text/input-text.js (hunk 2 above)
Comment (ngotchac) [inline js/src/dapps/tokenreg/Inputs/Text/input-text.js]: Yep, it doesn't, but modified it.

File: js/src/dapps/tokenreg/Tokens/Token/token.js (hunk 3 above)
Comment (ngotchac) [inline js/src/dapps/tokenreg/Tokens/Token/token.js]: Should I always use `const`?

File: js/src/dapps/tokenreg/Tokens/actions.js (hunk 4 above)
Comment (jacogr) [inline js/src/dapps/tokenreg/Tokens/actions.js]: Perfect.

File: js/src/dapps/tokenreg/Tokens/Token/token.js (hunk 3 above)
Comment (jacogr) [inline js/src/dapps/tokenreg/Tokens/Token/token.js]: Yes. Unless we need to modify things. The best approach - just use const, if you assign something and it complains, let it :)

**Pull request merged**